x-pack/plugins/infra/server/routes/infra/lib/mapper.ts

```diff
@@ -32,16 +34,29 @@ export const mapToApiResponse = (
     };
   }
 
-  const hosts = buckets.map((bucket) => {
-    const metrics = convertMetricBucket(params, bucket);
-    const metadata = convertMetadataBucket(bucket);
+  const hosts = buckets
+    .map((bucket) => {
+      const metrics = convertMetricBucket(params, bucket);
+      const metadata = convertMetadataBucket(bucket);
 
-    return { name: bucket.key as string, metrics, metadata };
-  });
+      return { name: bucket.key as string, metrics, metadata };
+    })
+    .sort((a, b) => {
+      const cpuAValue = a.metrics.find((metric) => metric.name === 'cpu')?.value ?? null;
+      const cpuBValue = b.metrics.find((metric) => metric.name === 'cpu')?.value ?? null;
+
+      // If both cpuAValue and cpuBValue are not null or undefined, perform the comparison
+      return (cpuBValue! ?? 0) - (cpuAValue! ?? 0);
```

Comment:
Why do we need `!` here when we have the `?? 0` in place and the cpu value is `number | null`(maybe I am missing something here)

Reply:
You're correct, we don't need the nullish assertion